src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline

```diff
@@ -0,0 +1,494 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generated code. DO NOT EDIT!
+
+using Google.Protobuf.WellKnownTypes;
+using Google.Tagger.V1;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Google.Example.Library.V1.Snippets
+{
+
+    public class LibraryServiceClientSnippets
+    {
+        public async Task CreateShelfAsync()
+        {
+            // Snippet: CreateShelfAsync(Shelf,CallSettings)
+            // Additional: CreateShelfAsync(Shelf,CancellationToken)
+            LibraryServiceClient client = await LibraryServiceClient.CreateAsync();
+            Shelf shelf = new Shelf();
+            Shelf response = await client.CreateShelfAsync(
+                shelf);
+            // End snippet
+        }
+
+        public void CreateShelf()
+        {
+            // Snippet: CreateShelf(Shelf,CallSettings)
+            LibraryServiceClient client = LibraryServiceClient.Create();
+            Shelf shelf = new Shelf();
+            Shelf response = client.CreateShelf(
+                shelf);
+            // End snippet
+        }
+
+        public async Task GetShelfAsync()
+        {
+            // Snippet: GetShelfAsync(string,CallSettings)
+            // Additional: GetShelfAsync(string,CancellationToken)
+            LibraryServiceClient client = await LibraryServiceClient.CreateAsync();
+            string name = "";
+            Shelf response = await client.GetShelfAsync(
+                name);
+            // End snippet
+        }
+
+        public void GetShelf()
+        {
+            // Snippet: GetShelf(string,CallSettings)
+            LibraryServiceClient client = LibraryServiceClient.Create();
+            string name = "";
+            Shelf response = client.GetShelf(
+                name);
+            // End snippet
+        }
+
+        public async Task GetShelfAsync()
+        {
+            // Snippet: GetShelfAsync(string,SomeMessage,CallSettings)
+            // Additional: GetShelfAsync(string,SomeMessage,CancellationToken)
+            LibraryServiceClient client = await LibraryServiceClient.CreateAsync();
+            string name = "";
+            SomeMessage message = new SomeMessage();
+            Shelf response = await client.GetShelfAsync(
+                name, message);
+            // End snippet
+        }
+
+        public void GetShelf()
```

Comment:
We appear to have two GetShelf and two GetShelfAsync methods (in the snippets file). Presumably this is due to multiple parameter flattenings... you could either just label them with a number, or put the parameters in the method name.